GNU debugger does not work well with Korn shell, it
either fails with the fatal assertion or breakpoints
are completely ignored. To workaround Korn shell issues,
we force GDB to internally use well-tested bash shell.

Not suitable for upstream.

--- gdb-15.1/gdbsupport/pathstuff.cc
+++ gdb-15.1/gdbsupport/pathstuff.cc
@@ -372,11 +372,7 @@ /* See gdbsupport/pathstuff.h.  */
 const char *
 get_shell ()
 {
-  const char *ret = getenv ("SHELL");
-  if (ret == NULL)
-    ret = "/bin/sh";
-
-  return ret;
+  return "/bin/bash";
 }
 
 /* See gdbsupport/pathstuff.h.  */
